tapchannel/aux_traffic_shaper.go

```diff
@@ -308,11 +310,20 @@ func (s *AuxTrafficShaper) ProduceHtlcExtraData(totalAmount lnwire.MilliSatoshi,
 	)
 	numAssetUnits := numAssetUnitsFp.ScaleTo(0).ToUint64()
 
-	// We now know how many units we need. We take the asset ID from the
-	// RFQ so the recipient can match it back to the quote.
-	assetId, err := quote.Request.AssetSpecifier.UnwrapIdOrErr()
-	if err != nil {
-		return 0, nil, fmt.Errorf("quote has no asset ID: %w", err)
+	var assetId asset.ID
+
+	switch {
+	case quote.Request.AssetSpecifier.HasId():
+		assetId = *quote.Request.AssetSpecifier.UnwrapIdToPtr()
+
+	case quote.Request.AssetSpecifier.HasGroupPubKey():
+		// If a group key is defined in the quote we place the hash of
+		// the group key as the dummy asset ID in the HTLC. This asset
+		// balance in the HTLC is just a hint and the actual asset IDs
+		// will be picked later in the process.
+		groupKey := quote.Request.AssetSpecifier.UnwrapGroupKeyToPtr()
+		groupKeyX := schnorr.SerializePubKey(groupKey)
```

Comment:
Commit says hash, but we use a serialized pubkey here.

Reply:
On the decoding side, do we take any steps to ensure that when we try to figure out if it's an asset ID or group key, we don't mistakenly think an asset ID is actually a group key? To elaborate a bit, there's a small probability that a given hash is actually a valid x-coord. This is how the NUMS key derivation works. 

I guess we can check our local DB or some other source to verify that it's actually a group key or asset ID?